src/cmds/fs/cat.c

```diff
@@ -43,26 +44,31 @@ int main(int argc, char **argv) {
 		}
 	}
 
-	if (NULL == (fd = fopen(argv[argc - 1], "r"))) {
-		printf("Can't open file %s\n", argv[argc - 1]);
-		return -errno;
-	}
+	for (i = 1; i < argc; i++) {
+		if (strcmp("-n", argv[i]) != 0) {
```

Comment:
Do you use getopt or manually parse arguments?

Reply:
There may be scenarios like 'cat file1 file2 -n' or 'cat file1 -n file2'. To handle these kinds of scenarios I am parsing manually and ignoring '-n' in 'argv'.

Reply:
It's not widely known, but this is actually an extension of GNU coreutils.
If you refer to POSIX (Utility Syntax Guidelines)[https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/V1_chap12.html#tag_12_02], then you can see
```
Guideline 9:
    All options should precede operands on the command line.
```
so POSIX-compatible `cat` utility would interpret `cat f1 -n` like two files.

But for me it's fine to deal with this option this way, so let's leave it as is.

Reply:
Oh! Thank you very much for the information.